pkg/api/types.go

```diff
@@ -1407,6 +1411,69 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimits is a list of rate limits on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimits []RateLimitSpec `json:"ratelimits,omitempty"`
```

Comment:
How about: 
1) remove DisruptionBudget
2) rename DisruptionBudgetSpec as DisruptionBudgetStrategy, and put it in Service and Job(use Pod annotation in the first version). 
3) Put RateLimitStatuses, AvailableReplicas, UnavailableReplicas, etc in ServiceStatus and JobStatus, directly. Do not use nested definition.